### DISCUSSION TITLE
Allow more template parameter for get_animal_view

### DISCUSSION THREAD

**Discussion by jtinsky** [sheltermanager/asm3#1580] [Q&A / Help]
Ideally I'd be able to use the template parameter for both adoptable and unadoptable animals in the /publishers/html->get_animal_view method copied below as a way of showing the detail view of a permanent foster to compliment the get_permfoster_animals code and having the rendered HTML match our existing site as much as possible without needing to alter the default "animalviewnotadoptable" template. 

```
def get_animal_view(dbo: Database, animalid: int, style="animalview") -> str:
    """ Constructs the animal view page to the animalview (or another specified) style/template. """
    a = dbo.first_row(get_animal_data(dbo, animalid=animalid, include_additional_fields=True, strip_personal_data=False))
    # The animal is adoptable, use the specified template
    if a is not None and is_animal_adoptable(dbo, a):
        head, body, foot = asm3.template.get_html_template(dbo, style)
        if head == "":
            head, body, foot = get_animal_view_template(dbo)
    else:
        # The animal is not adoptable. 
        # If there is no animalviewnotadoptable template, produce an error
        head, body, foot = asm3.template.get_html_template(dbo, "animalviewnotadoptable")
        if head == "": raise asm3.utils.ASMPermissionError("animal is not adoptable")
        # Otherwise, load the animal record so we can generate the animalviewnotadoptable page
```

**Comment by bobintetley** [comment]
Ok, I see what you're getting at. It does make sense. I'll put this on our dev list.

**Comment by jtinsky** [comment]
That would definitely work and probably be more flexible going forward too. On Dec 1, 2024, at 1:34 AM, Robin Rawson-Tetley ***@***.***> wrote:﻿
Ok, I see what you're getting at. It does make sense. I'll put this on our dev list.
